flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/stream/StreamingMultiJoinOperator.java

```diff
@@ -614,6 +615,16 @@ private Integer processRecords(
         }
     }
 
+    private static RowData newJoinedRowData(int depth, RowData joinedRowData, RowData record) {
+        RowData newJoinedRowData;
+        if (depth == 0) {
+            newJoinedRowData = record;
+        } else {
+            newJoinedRowData = new JoinedRowData(joinedRowData, record);
```

Comment:
Should we introduce a `MultiJoinedRowData` instead of having a chain of nested ones?

Reply:
Yes, it's a good idea. I suggest we do it in another ticket, created it here https://issues.apache.org/jira/browse/FLINK-37917

The reason for that is that it's basically an optimization and could be addressed after the first e2e working version. Wdyt?